### ISSUE TITLE
(3.0.53) RowanClassServiceTest>>#test_setSuperSubIndicators (commit 76b27b8) failing

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#379]
Here are the test failures I just noticed:
```
Rowan STON Cypress Tonel tests for GemStone '3.2.15'
648 run, 637 passed, 5 failed, 6 errors
  errors
-->	RowanClassServiceTest>>#test_setSuperSubIndicators
-->	RowanTestClassServiceTest>>#test_setSuperSubIndicators
	RwBrowserToolApiTest>>#testMoveGlobalExtensionSessionMethods
	RwHybridBrowserToolTest>>#testHybridMoveMethodFromSessionMethodsIntoSessionMethods
	RwHybridBrowserToolTest>>#testHybridMoveMethodIntoSessionMethods
	RwRowanProjectIssuesTest>>#testIssue150_branches
  failures
	RowanClassServiceTest>>#test_addedProjectNotOnDisk
	RowanProjectServiceTest>>#test_addedProjectNotOnDisk
	RowanTestClassServiceTest>>#test_addedProjectNotOnDisk
	RwLoadingTest>>#testPoolDictionaryChanges
	RwSymbolDictionaryTest>>#testClassVersioningPatch
```
[When I did the original merge of Oscar-3.0.53](https://github.com/GemTalk/Rowan/commit/b89d8e9089cdb7086036cad5e864ca20711e38e4) the tests passed ... Also [when I did my most recent commit on the candidateV1.2 branch](https://github.com/GemTalk/Rowan/commit/76b27b89b59d0dac82d2fd6458b1c01ac0845125) the tests pass ... but today running in a stone built in a new location the tests fail in a batch job run against both v1.3 and v1.2  ... as well as failing interactivley .. 

Like #333 and #336, the test failures should be addressed for the January release ... 

In the interactive failure the class **RowanSubClass** is missing and a search for the string in the image, shows that the only place this string shows up is in the test itself ... I'm guessing that this test is depending upon side effects from other tests to run correctly ...

**Comment by ericwinger** [comment]
> Like #333 and #336, the test failures should be addressed for the January release ...

@dalehenrich - did you mean #335?  #336 is not related to a failing test.

**Comment by dalehenrich** [comment]
Sorry, I was referring to #366, not #336

**Comment by ericwinger** [comment]
@dalehenrich The test is relying on the ``RowanSample1`` project which I load. ``RowanSubClass`` is in that project. Is it not safe to assume that RowanSample1 will be available as a directory when the server tests are run? 

```
	RowanProjectService new newGitProject: 'file:$ROWAN_PROJECTS_HOME/Rowan/samples/RowanSample1.ston' root: '$ROWAN_PROJECTS_HOME' useSsh: true. 
	Rowan projectTools load loadProjectNamed: 'RowanSample1'.
```

**Comment by dalehenrich** [comment]
The two lines of code you show will certainly create (if needed) and load the RowanSample1 project, however it is not a good practice for tests to download projects into `$ROWAN_PROJECTS_HOME` ... you run the risk of installing a project into a user's space and potentially disrupting the user's own work ... I think what happened in this case is that you starting using my local version of RowanSample1 and it so happens that I don't have my repo on a branch that you were expecting so the tests are failing ... 

A better practice for tests is to use the test scratch directory for installing test projects: `Rowan/test/testRepositories/repos` and using [this test code as a template for installing your repositories](https://github.com/GemTalk/Rowan/blob/candidateV1.2/rowan/src/Rowan-Tests/RwRowanSample4Test.class.st#L90-L111) ... This way the test code is under your total control and cannot affect the users own `ROWAN_PROJECTS_HOME`.

**Comment by ericwinger** [comment]
@dalehenrich  Ok, I'll take a stab at installing RowanSample1 using your suggested code.

**Comment by ericwinger** [comment]
@dalehenrich  Trying to implement [the code you referred to ](https://github.com/GemTalk/Rowan/blob/candidateV1.2/rowan/src/Rowan-Tests/RwRowanSample4Test.class.st#L90-L111) ...

This line of code (shown below in the debugger) appears to expect a ``.ston`` file which does not exist in the ``RowanSample1`` repository. Is it generated in the rowan tests somehow?  Or should I be pulling RowanSample1 from somewhere other than https://github.com/dalehenrich/RowanSample1

![image](https://user-images.githubusercontent.com/36212516/50669640-35d42380-0f7b-11e9-9d33-c62332924e7b.png)

**Comment by dalehenrich** [comment]
It looks like the path in the debugger is referring to [this directory in the RowanSample1](https://github.com/dalehenrich/RowanSample1/tree/master/rowan/specs) and there are two `.ston` files to choose from ... I'd suggest this one: `RowanSample1.ston`  ...

**Comment by ericwinger** [comment]
That seemed, at first glance, to get my over the hurdle.

**Comment by ericwinger** [comment]
RowanSample1 load updated in test & subclass created on fly to avoid dependency on disk version of project. Should be fixed in Oscar 3.0.54 but I'll double check before releasing.

**Comment by ericwinger** [comment]
Verified after integrating with CandidateV1.2 that this test no longer fails.
